Kudu.Core/Deployment/DeploymentManager.cs

```diff
@@ -293,12 +294,24 @@ public void Delete(string id)
             }
         }
 
-        public async Task RestartMainSiteIfNeeded(ITracer tracer, ILogger logger)
+        public async Task RestartMainSiteIfNeeded(ITracer tracer, ILogger logger, DeploymentInfoBase deploymentInfo)
         {
             // If post-deployment restart is disabled, do nothing.
             if (!_settings.RestartAppOnGitDeploy())
             {
                 return;
+            }
+
+            // Proceed only if 'restart' is allowed for this deployment
+            if (deploymentInfo != null && !deploymentInfo.RestartAllowed)
```

Comment:
Is `RestartAllowed` added as a OneDeploy only property ?

Reply:
No. But OneDeploy is the only scenario that changes the default. For everyone else, `RestartAllowed=true` (default behavior)